### PR TITLE
simplify code and update vendor

### PR DESCRIPTION
**Pull request by pete911**
there are a lot of changes, mainly in vendor because I had to update it, otherwise I wouldn't be able to run the code. The main changes are in `out_newrelic.go` and `nrclient.go` the rest is just formatting or dependencies.
Client now returns retry and error, making it easier for output to decide when to retry. Logic is not split between 2 separate components anymore.

### CODE REVIEW

File: nrclient/nrclient.go
Comment (danybmx) [inline nrclient/nrclient.go:63]: I think this can be extracted to a method named `isStatusCodeRetryable` for example, that returns true or false. It will improve the readability on this method and also separate the responsibility of check if we should retry or not for a given status code from the `Send` method.

File: nrclient/nrclient.go
Comment (danybmx) [inline nrclient/nrclient.go:60]: I would prefer to have a `if statusCode/100 != 2` and inside it, the logic of returning `true/false, nil` instead the `continue`.

This is just to improve readability of the code and avoid need to read carefully that there is a `continue` and other returns at the end of the for statement will be not executed. 

Mixing this, with the other comment, could be something like:

```
	for _, payload := range payloads {
		statusCode, err := nrClient.sendPacket(payload)
		if err != nil {
			return false, err
		}
		if statusCode/100 != 2 {
			return isStatusCodeRetryable(statusCode), nil
		}
	}
```

**Review by danybmx — COMMENTED** [review]
Hi @pete911, first of all, thanks for your contribution!! kudos!

I'm going to take a look more in depth and do some testing, but I already leave some suggestions there, could you take a look?

Regards and thanks again!

**Comment by pete911** [comment]
Hi @danybmx I have updated the code as you suggested

**Review by danybmx — CHANGES_REQUESTED** [review]
Hi @pete911 thanks a lot for this,

Only one last thing before merge, could you update the VERSION in `version.go` to `1.12.0`?

Thanks and regards! Kudos!

**Comment by pete911** [comment]
> Hi @pete911 thanks a lot for this,
> 
> Only one last thing before merge, could you update the VERSION in `version.go` to `1.12.0`?
> 
> Thanks and regards! Kudos!

done

**Comment by danybmx** [comment]
Hi @pete911,

Apologies, we're working on some changes in the pipeline, so we can't merge this until they're done.

We really appreciate your work and time so just want to give some feedback about why this is not already merged.

Regards

**Comment by pete911** [comment]
No worries @danybmx, let me know when it is merged, I have some more PRs in the pipeline, but wanted to make this "cleanup" first so I'm not introducing too many changes. Thanks for letting me know.

**Review by danybmx — APPROVED** [review]

**Comment by danybmx** [comment]
Hi @pete911,

We have this already merged and release + docker images published! 🎉  thanks!

We also discovered that unit tests are not reporting the fails on the pipeline for forked PR's and that the unit tests were failing after your changes, so we adapted the tests in #98 and also fixed the issue. Now, unit tests errors will be blocking PRs from being merged even for external forks.

Regards and thanks again!

**Pull request merged**
